rbc/omnisci_backend/omnisci_buffer.py

```diff
@@ -173,10 +180,39 @@ def omnisci_buffer_getitem(x, i):
         return lambda x, i: omnisci_buffer_getitem_(x, i)
 
 
+# [rbc issue-197] Numba promotes operations like
+# int32(a) + int32(b) to int64
+def truncate_or_extend(builder, nb_value, eltype, value, buf_typ):
+    # buf[pos] = val
+
+    if isinstance(nb_value, types.Integer):  # Integer
+        if eltype.bitwidth < nb_value.bitwidth:
+            return builder.trunc(value, buf_typ)  # truncate
+        elif eltype.bitwidth > nb_value.bitwidth:
+            is_signed = nb_value.signed
+            return builder.sext(value, buf_typ) if is_signed else \
+                builder.zext(value, buf_typ)  # extend
+    elif isinstance(nb_value, types.Float):  # Floating-point
+        if eltype.bitwidth < nb_value.bitwidth:
+            return builder.fptrunc(value, buf_typ)  # truncate
+        elif eltype.bitwidth > nb_value.bitwidth:
+            return builder.fpext(value, buf_typ)  # extend
+    elif isinstance(nb_value, types.Boolean):
+        if buf_typ.width < value.type.width:
+            return builder.trunc(value, buf_typ)
+        elif buf_typ.width > value.type.width:
+            return builder.sext(value, buf_typ)
```

Comment:
I wonder in which circumstances are these returns effective?

Reply:
```python
@omnisci('bool[](bool[])')
def fn_issue197_bool(x):
    y = np.zeros_like(x)
    for i in range(len(x)):
        y[i] = bool(x[i])
    return y
```

In the test above, `bool` casts the value to `int1`.